substrate/frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1616,15 +1644,7 @@ impl<T: Config> Pallet<T> {
 	// - [`DesiredTargets`] exists if and only if [`Snapshot`] is present.
 	// - [`SnapshotMetadata`] exist if and only if [`Snapshot`] is present.
 	fn try_state_snapshot() -> Result<(), TryRuntimeError> {
-		if <Snapshot<T>>::exists() &&
-			<SnapshotMetadata<T>>::exists() &&
-			<DesiredTargets<T>>::exists()
-		{
-			Ok(())
-		} else if !<Snapshot<T>>::exists() &&
-			!<SnapshotMetadata<T>>::exists() &&
-			!<DesiredTargets<T>>::exists()
-		{
+		if SnapshotWrapper::<T>::exists() || SnapshotWrapper::<T>::not_exists() {
```

Comment:
The way the `fn exists` and `fn not_exists` are named make this assertion a bit funny: "if it exists OR it does not exist", which should always be true.

I think we can take a step back and think about what's being checked here. We want to check if the storage items in the wrapper are inconsistent or not. Maybe we could add a fn to the wrapper called `fn is_consistent` where all the checks are performed.